rpc/flipt/evaluation/evaluation.proto

```diff
@@ -38,13 +37,15 @@ enum EvaluationReason {
 enum EvaluationFlagType {
   VARIANT_FLAG_TYPE = 0;
   BOOLEAN_FLAG_TYPE = 1;
+  ERROR_FLAG_TYPE = 2;
```

Comment:
totally get why this is here, but looking at it.. it just seems weird to have an `error` `flag type`. I wonder if we should change `EvaluationResponse.flag_type` to `EvaluationResponse.type` so that it's clear that it's a field used to denote the `response` object type and not necessarily a type of flag.

long way of me saying, maybe we need a new `enum` `EvaluationResponseType` where the values are:

```proto
VARIANT_EVALUATION_RESPONSE_TYPE = 0;
BOOLEAN_EVALUATION_RESPONSE_TYPE = 1;
ERROR_EVALUATION_RESPONSE_TYPE = 2;
```

and the `EvaluationResponse` changes to:

```proto
message EvaluationResponse {
    EvaluationResponseType type = 1;
    oneof response {
        BooleanEvaluationResponse boolean_response = 2;
        VariantEvaluationResponse variant_response = 3;
        ErrorEvaluationResponse error_response = 4;
    }
}
```

wyt?

Reply:
@markphelps Yeah agreed. I know we removed flag type from the request, but yeah the flag type of error is not really of any material value. It makes sense to have it as a separate type.